### PR TITLE
Migration to typescript, take 2

### PR DESCRIPTION
**Pull request by Zaista**

### CODE REVIEW

**Comment by mrhyde** [comment]
@Zaista, thank you for addressing this issue. I truly appreciate your assistance. Please don't hesitate to reach out if you require any further help in completing this task.

**Comment by Zaista** [comment]
Np, in the meantime, I moved it to my fork of your fork: https://github.com/Zaista/cypress-visual-regression :D
And also opened a new PR
Thanks a lot for all the progress that you guys did so far 🚀

File: cypress.base.config.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { defineConfig } from 'cypress'
+import fs from 'fs'
+import getCompareSnapshotsPlugin from './src/plugin'
+
+export default defineConfig({
+  // screenshotsFolder: './cypress/snapshots/actual', TODO this doesn't work ?!
```
Comment (Zaista) [inline cypress.base.config.ts:6]: No idea atm why this doesn't work

File: src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+export type UpdateSnapshotArgs = {
+  screenshotName: string
+  specRelativePath: string
+  specFolder: string
+  screenshotsFolder: string
+  snapshotBaseDirectory: string
+}
+
+function getSpecTrimmedPath(relativePath: string, integrationFolder: string): string {
```
Comment (Zaista) [inline src/plugin.ts:22]: This part is the most troubling one:
 - two things to consider:
    1) path to specFolder (previously called integrationFolder (here still to be updated))
    2) spec nested positions (see sub/deep.spec.ts test file)
 These are different in run/open mode which drive me nuts, plus specFolder cannot be simply taken from specPattern as it's not a path
I've tried many different things just to end up with some edge case problems... I'll have to see if a spec absolute path can be used in a good way
